management/server/http/posture_checks_handler.go

```diff
@@ -342,3 +368,31 @@ func toPostureGeoLocationCheck(apiGeoLocationCheck *api.GeoLocationCheck) *postu
 		Locations: locations,
 	}
 }
+
+func toPrivateNetworkCheckResponse(check *posture.PrivateNetworkCheck) *api.PrivateNetworkCheck {
+	netPrefixes := make([]string, 0, len(check.Prefixes))
+	for _, netPrefix := range check.Prefixes {
+		netPrefixes = append(netPrefixes, netPrefix.String())
+	}
+
+	return &api.PrivateNetworkCheck{
+		Prefixes: netPrefixes,
+		Action:   api.PrivateNetworkCheckAction(check.Action),
+	}
+}
+
+func toPrivateNetworkCheck(check *api.PrivateNetworkCheck) (*posture.PrivateNetworkCheck, error) {
+	prefixes := make([]netip.Prefix, 0)
```

Comment:
we should check if the len of incoming ranges > 0 and return an error otherwise

Reply:
This is already checked during [validation](https://github.com/netbirdio/netbird/blob/d7db3e615cea8a24b9c4ae78a865e75d885e67a0/management/server/http/posture_checks_handler.go#L290).